### PR TITLE
NEW:Show units of products in stocks and hide total units if content is of diffents units

### PR DESCRIPTION
**Pull request by bafbes**
fix:Show units of products in stocks and hide total units if content is of diffents units

### CODE REVIEW

File: htdocs/product/stock/card.php [hunk 1]
```diff
@@ -620,15 +630,24 @@
 						print "</a></td>";
 					}
 
-					print "</tr>";
-					$i++;
-				}
-				$db->free($resql);
+                    if ($user->rights->stock->creer) {
```
Comment (eldy) [inline htdocs/product/stock/card.php]: It seems that the test on "if ($user->rights->stock->creer)" is already present just before. This if become duplicate after patch.

**Comment by bafbes** [comment]
Fixed unneeded tests

File: htdocs/product/stock/card.php (hunk 1 above)
Comment (eldy) [reply htdocs/product/stock/card.php]: This part of code is still duplicated (the if and its content is done twice)

Comment (bafbes) [reply htdocs/product/stock/card.php]: I hope to have succeeded in it
I also fixed a condition test and added display units type if products in warehouse are of same nature

File: htdocs/product/stock/card.php [hunk 2]
```diff
@@ -618,19 +628,27 @@
 						print '<td class="center"><a href="'.DOL_URL_ROOT.'/product/stock/product.php?dwid='.$object->id.'&id='.$objp->rowid.'&action=correction&backtopage='.urlencode($_SERVER["PHP_SELF"].'?id='.$id).'">';
 						print $langs->trans("StockCorrection");
 						print "</a></td>";
-					}
-
-					print "</tr>";
-					$i++;
-				}
-				$db->free($resql);
+                        print '<td align="center"><a href="' . DOL_URL_ROOT . '/product/stock/product.php?dwid=' . $object->id . '&id=' . $objp->rowid . '&action=correction&backtopage=' . urlencode($_SERVER["PHP_SELF"] . '?id=' . $id) . '">';
```
Comment (eldy) [inline htdocs/product/stock/card.php]: Still wrong.
The column "print $langs->trans("StockCorrection");" is now duplicated.

Comment (bafbes) [reply htdocs/product/stock/card.php]: Fixed

**Pull request merged**